compiler-core/src/format.rs

```diff
@@ -1091,9 +1103,23 @@ impl<'comments> Formatter<'comments> {
         args: &'a [SmolStr],
         constructors: &'a [RecordConstructor<A>],
         location: &'a SrcSpan,
+        deprecation: &'a Deprecation,
     ) -> Document<'a> {
         let _ = self.pop_empty_lines(location.start);
-        let doc = pub_(public)
+        let doc = docvec![];
+
+        // @deprecated attribute
+        let doc = match deprecation {
+            Deprecation::NotDeprecated => doc,
+            Deprecation::Deprecated { message } => doc
+                .append("@deprecated(\"")
+                .append(message)
+                .append("\")")
+                .append(line()),
+        };
```

Comment:
Could you add a test for the formatting of this please? Thanks

Reply:
Yep will add a test for this.